odoo/addons/base/models/ir_mail_server.py

```diff
@@ -259,6 +267,10 @@ def connect(self, host=None, port=None, user=None, password=None, encryption=Non
             else:
                 ssl_context = None
 
+        # Do not actually connect while running in test mode
+        if getattr(threading.currentThread(), 'testing', False):
```

Comment:
Meh. I really don't think this should be done like that. See above remarks about mock and patch to clean and sort out.

Reply:
We can not really mock, because we should do it for almost all tests in Odoo... That's why they do that (it's an existing code, I just moved it)